src/lightning_app/cli/lightning_cli.py

```diff
@@ -116,6 +118,83 @@ def run_app(
     _run_app(file, cloud, without_server, no_cache, name, blocking, open_ui, env)
 
 
+@main.group()
+def exec():
```

Comment:
I am not sure the name `exec` is "self explaning" what actually it does. The first impression I had is it executes any system command in the app, which is wrong but it actually runs a function of the app defined by a shortcut.

Reply:
Any other ideas there ? @awaelchli too ?

Reply:
maybe 
`lightning run function  --args name=<value> --app_id_or_name=<>`

`app_id_or_name` is a bit not friendly, maybe having separate `app_id` and `app_name`

Reply:
I would not use `run` here so that we really have totally different entrypoints for triggering functions and running apps. What about `trigger`?

Reply:
I like `trigger`, it's exactly what we do 'trigger a function'. But I think having a common entrypoint for 

- run an app `lightning run app my_app.py`
- run a function `lightning run function <my_function>` 

is a good thing.

Reply:
I would not go with `lightning run function ...` (or `command`) here's my train of though (allow me to pour myself here :P)
For `run app`:
- As a user I see `lightning run app my_app.py` - Knowing what it means if fairly self-explanatory because we assume that user already knows that lightning is for creating lightning apps. So that command just starts the application.
- On the other side. I'm a user and I have an app - but how do I start it? There should be a command for that - looks at the docs - finds something that relates to starting (i.e. `run` command) and I'm happy, I know what to do.

For the `run command` / `function`:
- As a user I see the command `run command ...` - I know that lightning is for running the apps, I did not know about commands / functions. They are under `run` so are they a separate entity that exists in the lightning framework? -- Bottom line we would have to explain the the help that you have to have an app running to execute that, and it's actually not really running anything but actually it's something like RPC (remote procedure call)
- From the other side: I see that the code: `configure_commands` suggest it's something that should be used for configuration. I know that we almost always have to configure a machine we want to do something on, so does that mean those commands will be called as hooks during configuration? Or maybe it's just like `configure_layout`? If so what are the `commands` here?

I think that conveying the meaning of that feature can be quite tricky. 
Side question, do we just execute the function, or we also can retrieve data from it to print to the terminal ? 

Personally I'd think about something like `lightning rpc <function_name> --args ...` (or if we do not like acronyms just `lightning call <function_name> ...`) because it mostly resembles my understanding of what this is actually for. It also suggests that it's a call, so we "implicitly" feel that it's not spinning (running) anything additional - so the app has to be already running?
Also I'd go change `configure_commands` to something like `remote_calls` ? `configure_remote_calls`? dunno tbh. We already have `configure_layout` so I imagine we'd like to keep the convention similar?

Reply:
On the other note. `run function` strongly suggest that we'd run something like AWS lambda function -- At least for me, I know it might be just because of my experience with AWS, but still...